imagersite/imager_images/views.py

```diff
@@ -1,3 +1,35 @@
 from django.shortcuts import render
+from django.core.urlresolvers import reverse
+from django.views.generic.edit import CreateView
+from .models import Album, Photo, AlbumForm, PhotoForm
+from django.http import HttpResponseRedirect
 
 # Create your views here.
+
+# class AddAlbumView(CreateView):
+#     model = Album
+#     # exclude = ['date_published', 'owner']
+#     fields = ['cover', 'title', 'description', 'published']
+#     template_name = "add_album.html"]
+
+
+def add_album(request):
+    form = AlbumForm()
+    if request.method == 'POST':
+        form = AlbumForm(request.POST, request.FILES)
```

Comment:
you don't really need to build this form twice.  It's true that `request.POST` will be an empty dict if the request that came in was a GET request, as will request.FILES, so there's no penalty for building the form once with all the things you need already in place.  Same goes for below.